crates/ironrdp-svc/src/lib.rs

```diff
@@ -126,81 +133,132 @@ pub trait StaticVirtualChannel: AsAny + fmt::Debug + Send + Sync {
 
 assert_obj_safe!(StaticVirtualChannel);
 
-/// Takes a vector of PDUs and breaks them into chunks prefixed with a Channel PDU Header (`CHANNEL_PDU_HEADER`).
-///
-/// Each chunk is at most `max_chunk_len` bytes long (not including the Channel PDU Header).
-pub fn chunkify(messages: Vec<SvcMessage>, max_chunk_len: usize) -> PduResult<Vec<WriteBuf>> {
-    let mut results = vec![];
-    for message in messages {
-        results.extend(chunkify_one(message, max_chunk_len)?);
-    }
-    Ok(results)
+/// SvcPreprocessor is used to chunkify/de-chunkify static virtual channel PDUs.
+#[derive(Debug)]
+pub struct SvcPreprocessor {
```

Comment:
I feel that `ChunkProcessor` or `SvcChunkProcessor` would be a somewhat more obvious name for this helper